torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -1174,12 +1081,12 @@ class ConcretizeDomain : private BackwardVisitor {
 
   // Returns true if either id is not a broadcast or
   // the traversal has found a concretized axis for id
-  bool canConcretize(IterDomain* id) const {
+  bool canConcretize(const IterDomain* id) const {
     return !id->isBroadcast() || bcast_domain_map_.count(id);
   }
 
   // Returns the concretized id recorded from traversal
-  IterDomain* concretized(IterDomain* id) const {
+  const IterDomain* concretized(const IterDomain* id) const {
```

Comment:
I guess you're asking about `tv1->getRootDomain()[1]` since the 0-th rood domain is not broadcast. In any case, `concretizeDomain` only returns a corresponding `IterDomain` that has a concrete size. Whether that identified concrete `IterDomain` has the same size as other `IterDomain`s is a different question. Pinging @shmsong as this was originally written by him.

Reply:
The assumption of "concretizing" a broadcast axes is "single use"/"single semantic" for the broadcast axes, and if the same broadcast axis is used in multiple places like in the example:
```cpp
auto tv3 = mul(tv1, tv2);
auto tv5 = mul(tv1, tv4);
```
and say
```
tv1 = T1[i0,b1]
tv2 = T2[i1,i2]
tv4 = T4[i3,i4]
```
Then i2 and i4 will be assumed equal, and actually the `IterDomain:: proveEquivalent` will return them as equal. 

One possible way around if tv2 and tv4 don't actually have the same dimensions is explicitly create a new broadcast for each use since they're actually different broadcast semantics (i.e. sizes).

Would be happy to refactor any part if this isn't the particular behavior that we want.

Reply:
I assumed that a single broadcast IterDomain must be uniquely mapped to a single IterDomain with a concrete size. That's likely the case most of the times, but not necessarily true. We should remove the assumption or fail if not.

Having a new broadcast would make things simpler, but that'd be a user-interface question. I think what should be done anyway is that we should be able to detect broadcast IterDomains that may be mapped to multiple different domains.

Reply:
Sure, detection of a `broadcast` mapping to multiple concrete `IterDomains` can be done on top of the `proveEquivalent` class. I can add that somewhere if we want. Meanwhile I was curious how this should be handled in a global root domain mapping, since multiple broadcast semantics will make the result of mapping unavoidably context-dependent.